pkgs/build-support/setup-hooks/auto-patchelf.sh

```diff
@@ -97,9 +102,14 @@ findDependency() {
 }
 
 autoPatchelfFile() {
-    local dep rpath="" toPatch="$1"
+    local dep interpreter rpath="" toPatch="$1"
+
+    if [ "$(getElfClass "$toPatch")" = "ELF32" ]; then
+        interpreter=$(< "@ld32@")
+    else
+        interpreter=$(< "@ld@")
+    fi
```

Comment:
I really don't like hardcoding this, as this is going to ask for trouble on non-Intel architectures. What about looping through `cat $NIX_CC/nix-support/dynamic-linker*` instead and compare the value of the interpreter with the outputs of `getSoArch`? This would also give us an error if we have an executable where no matching interpreter exists.

Reply:
Ah, this makes sense. Package authors would have to use `stdenv_32bit` to get multilib support, but it would remove the need to hardcode the interpreter paths per-architecture. I'll try this.